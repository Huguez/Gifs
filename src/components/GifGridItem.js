import React from 'react'
import { PropTypes } from 'prop-types';

export const GifGridItem = ( {imagen, title, id} ) => {

    // console.log( imagen );
    return (
        <div className="card animate__animated animate__fadeIn">
            <p> { title } </p>
            <img alt={title} src={ imagen } />
        </div>
    )
}

GifGridItem.prototype = {
    imagen: PropTypes.string.isRequired,
    title:  PropTypes.string.isRequired,
    id:     PropTypes.string
}
