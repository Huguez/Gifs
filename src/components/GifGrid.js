import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Proptypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );
    
    return ( <>
        <h3> { category } </h3>
        { loading && <p className="animate__animated animate__flash "> Loading... </p> }

        <div className="card-grid"> 
            { data.map( ( item ) => { 
                return <GifGridItem key={item.id} {...item } />
            } ) }
        </div> 
    </>);
}

GifGrid.prototype  = {
    category: Proptypes.string.isRequired
}