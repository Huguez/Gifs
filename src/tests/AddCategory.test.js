import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../components/AddCategory";

describe( 'Pruebas con AddCategory.js', ()=>{

    const setCategory = jest.fn();

    let wrapper = shallow( <AddCategory setCategories={ setCategory } /> );

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategory } /> );
    })

    test( 'debe de mostrar correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    } );

    test('debe de cambiar la caja de texto', ()=>{
        const input = wrapper.find( 'input' );
        const value = "hacker";
        input.simulate('change', { target: { value } } );

        expect( wrapper.find( 'p' ).text().trim()  ).toBe( value )

    } );

    test( 'No debe de postear la informacion', () => {
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } );

        expect( setCategory ).not.toHaveBeenCalled();
    } );

    test(' Debe de llamar el setCategory ', ()=>{
        
        const value = "hacker";
        
        wrapper.find( 'input' ).simulate('change', { target: { value } } );
        
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} } ); 
                
        expect( setCategory ).toHaveBeenCalled();
        expect( setCategory ).toHaveBeenCalledTimes(1);
        expect( setCategory ).toHaveBeenCalledWith( expect.any( Function ) )
        
        expect( wrapper.find( 'input' ).props().value  ).toBe( '' );
    } );
    
} );