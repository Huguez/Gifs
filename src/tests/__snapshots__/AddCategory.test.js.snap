// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con AddCategory.js debe de mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="categoria"
    type="text"
    value=""
  />
</form>
`;
